jupyter_core/tests/test_paths.py

```diff
@@ -139,39 +139,50 @@ def test_data_dir_env():
     assert data == data_env
 
 
-@pytest.mark.skipif(sys.platform == "win32", reason="does not run on windows")
+@pytest.mark.skipif(sys.platform != "darwin", reason="only run on osx")
 def test_data_dir_darwin():
-    with darwin:
-        data = jupyter_data_dir()
+    data = jupyter_data_dir()
     assert data == realpath("~/Library/Jupyter")
 
-    with darwin, xdg:
+    with xdg:
         # darwin should ignore xdg
         data = jupyter_data_dir()
-    assert data == realpath("~/Library/Jupyter")
+        assert data == realpath("~/Library/Jupyter")
+
+    with patch.dict("os.environ", {"JUPYTER_PLATFORM_DIRS": "1"}):
+        data = jupyter_data_dir()
+        assert data == realpath("~/Library/Application Support/Jupyter")
 
 
 @pytest.mark.skipif(sys.platform != "win32", reason="only run on windows")
 def test_data_dir_windows():
     data = jupyter_data_dir()
     assert data == realpath(pjoin(os.environ.get("APPDATA", ""), "jupyter"))
 
-    with xdg:
-        # windows should ignore xdg
+    with patch.dict("os.environ", {"JUPYTER_PLATFORM_DIRS": "1"}):
         data = jupyter_data_dir()
-    assert data == realpath(pjoin(os.environ.get("APPDATA", ""), "jupyter"))
+        assert data.endswith(r"AppData\Local\Jupyter")
 
 
-@pytest.mark.skipif(sys.platform == "win32", reason="does not run on windows")
+@pytest.mark.skipif(sys.platform != "linux", reason="only run on linux")
 def test_data_dir_linux():
-    with linux, no_xdg:
+    with no_xdg:
         data = jupyter_data_dir()
     assert data == realpath("~/.local/share/jupyter")
 
-    with linux, xdg:
+    with xdg:
         data = jupyter_data_dir()
     assert data == pjoin(xdg_env["XDG_DATA_HOME"], "jupyter")
 
+    with patch.dict("os.environ", {"JUPYTER_PLATFORM_DIRS": "1"}):
+        data = jupyter_data_dir()
+    assert data == realpath("~/.local/share/jupyter")
+
+    with patch.dict("os.environ", {"JUPYTER_PLATFORM_DIRS": "1", "XDG_DATA_HOME": "/tmp/xdg/data"}):
+        data = jupyter_data_dir()
+    # TODO: Should this be titlecase "Juptyer" by xdg convention?
+    assert data == pjoin(xdg_env["XDG_DATA_HOME"], "jupyter")
```

Comment:
Based on this list, I think it could go either way: https://wiki.archlinux.org/title/XDG_Base_Directory

Reply:
I think we ought to make it lowercase, then, for backwards compatibility, and to simplify that on linux, it is lowercase.

Reply:
Makes sense to me